### PR TITLE
Improve analysis event dispatch throughput

### PR DESCRIPTION
**Pull request by nscuro**
### Description

This PR is an attempt of increasing throughput when dispatching events for vulnerability, or repository metadata analysis for all components in the portfolio.

* Query data using "dumb" projections; Saves some ORM overhead
* Fetch components in batches of 5000 instead of 500; Saves lots of network calls
  * Memory impact should be minimal as projection records are tiny

Also added the `/api/v1/finding/portfolio/analyze` REST endpoint to manually trigger a portfolio analysis.

### Addressed Issue

N/A

### Additional Details

N/A

### Checklist

- [x] I have read and understand the [contributing guidelines](../CONTRIBUTING.md#pull-requests)
- ~This PR fixes a defect, and I have provided tests to verify that the fix is effective~
- [x] This PR implements an enhancement, and I have provided tests to verify that it works as intended
- ~This PR introduces changes to the database model, and I have added corresponding [update logic](https://github.com/DependencyTrack/dependency-track/tree/master/src/main/java/org/dependencytrack/upgrade)~
- ~This PR introduces new or alters existing behavior, and I have updated the [documentation](https://github.com/DependencyTrack/dependency-track/tree/master/docs/_docs) accordingly~

### CODE REVIEW

**Review by sahibamittal — APPROVED** [review]
lgtm 👍🏼

**Review by mehab — APPROVED** [review]

File: src/main/java/org/dependencytrack/tasks/RepositoryMetaAnalyzerTask.java [hunk 1]
```diff
@@ -100,23 +96,26 @@ private void processPortfolio() throws Exception {
...
             }
         }
 
         LOGGER.info("All components in portfolio submitted for repository meta analysis");
     }
 
-    private List<Component> fetchNextComponentsPage(final PersistenceManager pm, final Project project, final Long lastId) throws Exception {
+    private void dispatchComponents(final List<ComponentProjection> components) {
+        for (final var component : components) {
+            kafkaEventDispatcher.dispatchAsync(new ComponentRepositoryMetaAnalysisEvent(component.purl(), component.internal()));
```
[the remark this answers is not in the document]
Comment (nscuro) [reply src/main/java/org/dependencytrack/tasks/RepositoryMetaAnalyzerTask.java:114]: @sonatype-lift ignore

File: src/main/java/org/dependencytrack/tasks/VulnerabilityAnalysisTask.java [hunk 2]
```diff
@@ -112,24 +107,27 @@ private void processPortfolio(final UUID scanToken) throws Exception {
...
             }
         }
 
         LOGGER.info("All components in portfolio submitted for vulnerability analysis");
     }
 
-    private List<Component> fetchNextComponentsPage(final PersistenceManager pm, final Project project, final Long lastId) throws Exception {
+    private void dispatchComponents(final UUID scanToken, final List<ComponentProjection> components) {
+        for (final var component : components) {
+            eventDispatcher.dispatchAsync(new ComponentVulnerabilityAnalysisEvent(scanToken,
```
[the remark this answers is not in the document]
Comment (nscuro) [reply src/main/java/org/dependencytrack/tasks/VulnerabilityAnalysisTask.java:124]: @sonatype-lift ignore

File: src/main/java/org/dependencytrack/tasks/RepositoryMetaAnalyzerTask.java [hunk 3]
```diff
@@ -100,23 +96,26 @@ private void processPortfolio() throws Exception {
 
         try (final QueryManager qm = new QueryManager()) {
             final PersistenceManager pm = qm.getPersistenceManager();
-            pm.getFetchPlan().setGroup(Component.FetchGroup.REPOSITORY_META_ANALYSIS.name());
 
-            List<Component> components = fetchNextComponentsPage(pm, null, null);
+            List<ComponentProjection> components = fetchNextComponentsPage(pm, null, null);
```
[the remark this answers is not in the document]
Comment (nscuro) [reply src/main/java/org/dependencytrack/tasks/RepositoryMetaAnalyzerTask.java:100]: @sonatype-lift ignore

File: src/main/java/org/dependencytrack/tasks/VulnerabilityAnalysisTask.java [hunk 4]
```diff
@@ -112,24 +107,27 @@ private void processPortfolio(final UUID scanToken) throws Exception {
 
         try (final QueryManager qm = new QueryManager()) {
             final PersistenceManager pm = qm.getPersistenceManager();
-            pm.getFetchPlan().setGroup(Component.FetchGroup.VULNERABILITY_ANALYSIS.name());
-
-            List<Component> components = fetchNextComponentsPage(pm, null, null);
+            List<ComponentProjection> components = fetchNextComponentsPage(pm, null, null);
```
[the remark this answers is not in the document]
Comment (nscuro) [reply src/main/java/org/dependencytrack/tasks/VulnerabilityAnalysisTask.java:110]: @sonatype-lift ignore

File: src/main/java/org/dependencytrack/tasks/RepositoryMetaAnalyzerTask.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (nscuro) [reply src/main/java/org/dependencytrack/tasks/RepositoryMetaAnalyzerTask.java:114]: @sonatype-lift ignoreall

**Pull request merged**